src/main/java/org/onedatashare/server/module/http/HttpResource.java

```diff
@@ -0,0 +1,334 @@
+package org.onedatashare.server.module.http;
+
+import org.apache.commons.vfs2.*;
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.onedatashare.server.model.core.Resource;
+import org.onedatashare.server.model.core.Slice;
+import org.onedatashare.server.model.core.Stat;
+import org.onedatashare.server.model.core.Tap;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class HttpResource extends Resource<HttpSession, HttpResource> {
+    private String uri;
+
+    protected HttpResource(HttpSession session, String uri) {
+        super(session);
+        this.uri = uri;
+    }
+
+    @Override
+    public Mono<HttpResource> select(String path) {
+        return null;
+    }
+
+    public Mono<Stat> stat() {
+        return Mono.just(onStat());
+    }
+
+
+    /**
+     * This method returns the information parsed from the given http file server web page
+     * @return
+     * Returns the stat object
+     */
+    public Stat onStat() {
+        Stat stat = new Stat();
+
+        // Get the hostname from the uri
+        stat.setName(URI.create(uri).toString());
+
+        Document document;
+
+        // Fetch the document
+        try {
+            document = Jsoup.connect(uri).get();
+            document.select("th").remove();
+        } catch (IOException e) {
+            return stat;
+        }
+
+        // Select the table rows
+        Elements table = document.select("tr");
+        Stat contentStat;
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        if (table.size() > 0) {
+            ArrayList<Stat> contents = new ArrayList<>(table.size());
+            for (Element row : table) {
+                // Filter for removing queries, no links and links to parent directory
+                // Remove Query Strings or no links
+                if (row.toString().contains("href=?") || !row.toString().contains("href"))
+                    continue;
```

Comment:
Did not understand this check. Probably comments could be more specific.

Reply:
Added better comments